File: src/main/java/com/poker/poker/models/game/HandModel.java
package com.poker.poker.models.game;

import io.swagger.v3.oas.annotations.media.Schema;
import java.util.UUID;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class HandModel {
  @Schema(
      description = "Hand ID.",
      example = "0a7d95ef-94ba-47bc-b591-febb365bc543",
      implementation = UUID.class)
  @Id
  private UUID id;

  @Schema(
      description = "Game ID.",
      example = "0a7d95ef-94ba-47bc-b591-febb365bc543",
      implementation = UUID.class)
  private UUID gameId;

  // Temporary
  @Schema(
      description = "Temporary message.",
      example = "Player X won.",
      implementation = String.class)
  private String message;
}

File: src/app/api/models/hand-model.ts
/* tslint:disable */
import { HandActionModel } from './hand-action-model';
export interface HandModel {
  actions?: Array<HandActionModel>;

  /**
   * Game ID.
   */
  gameId?: string;

  /**
   * Hand ID.
   */
  id?: string;

  /**
   * Temporary message.
   */
  message?: string;

  /**
   * ID of the player whose turn it is.
   */
  playerToAct?: string;
}
